setup.py

```diff
@@ -70,10 +70,11 @@ def read(*names, **kwargs):
     license='AGPLv3+',
     install_requires=[
         'attrs',
+        'dataclasses; python_version=="3.6"',
         'keyring',
         'matplotlib',
         'notebook',
-        'numpy <1.16, >=1.12',
+        'numpy < 1.16, >= 1.12',
```

Comment:
@Centrus007 I was hoping to have a different work around for this upper limit. But it was taking too long to solve. So we'll have this upper limit for a bit longer.